### ISSUE TITLE
[bug] New virtualrunenv template: $: bad variable name

### ISSUE THREAD

**Issue by keef-cognitiv** [conan-io/conan#6264]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 19.10
  * Compiler+version: gcc 9 (didn't get to compilation)
  * Conan version: 1.21.0
  * Python version: 3.7.5rc1

### Steps to reproduce (Include if Applicable)
use `generators = "virtualbuildenv", "virtualrunenv"`

and then in the recipe:
`self.run(". ./activate_run.sh")`

Or really run activate_run.sh in sh (not bash)

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
<package>: Calling build()
/bin/sh: 10: export: $: bad variable name
```
The generated shell script for activate_run is not valid in vanilla sh... but self.run() seems to call `/bin/sh -c `

Content of file:
```

#!/usr/bin/env sh
export CONAN_OLD_PATH="$PATH"

while read line; do
    LINE="$(eval echo $line)";
    export "$LINE";
done < "/home/karagon/.conan/data/libpqxx/6.4.5/kmaragon/stable/build/5454c8cd87e600a4ac84733471af024e0c2da401/environment_run.sh.env"

export CONAN_OLD_PS1=$PS1
export PS1="(conanrunenv) $PS1"
```

Manually adding quotes around `$PS1` after `CONAN_OLD_PS1` fixes the issue.

**Comment by jgsogo** [comment]
Hi! It looks like this is something we can improve. Thanks for submitting the issue.

I would like to investigate the issue and add a test to reproduce the error in `develop` branch (maybe we should change our tests and use more vanilla sh), that's one task, but we can consider the fix proposed by @kmaragon in the PR for the next minor.

Thanks a lot!

**Comment by jgsogo** [comment]
Hi again @keef-cognitiv. 

I'm trying to reproduce this issue for a very simple use case to add a test in our CI but I'm not able to. Although I think that the patch is safe to apply this is something I would like to have in the CI to avoid any regression. All the Unix I find have `bash`, where can I get a vanilla shell?

---

On the other hand, I would like to know why are you calling `self.run(". ./activate_run.sh")`, and if the issue remains using `RunEnvironment` or the recommended `self.run("command", run_environment=True)` (see [docs here](https://docs.conan.io/en/latest/reference/build_helpers/run_environment.html#runenvironment))

Thanks!

**Comment by keef-cognitiv** [comment]
Sorry - I actually hadn't dug deeper into what Ubuntu uses. Rather than being a classic bourne shell, Ubuntu uses dash for /bin/sh. So the issue isn't with classic shell, but with Debian's variant Debian Almquist shell. So any base with Debian with the symlink of /bin/sh to dash will do it. The shell seems to target POSIX compliance with nothing more - hence this issue.

**Comment by keef-cognitiv** [comment]
And thanks for the tip re run_environment. I think that is enough for a functional work-around.

**Comment by jgsogo** [comment]
Hi, @keef-cognitiv! If the `self.run("command", run_environment=True)` works for you (not a workaround but the builtin and recommended way) then I prefer to move this issue to version `1.22` and try to reproduce it (see if it is possible in our CI) and provide a solution with tests.

wdyt? Is it ok for you if we postpone the investigation for the next release?

**Comment by keef-cognitiv** [comment]
If I'm the only one with the issue, that's 100% ok with me. run_environment was absolutely sufficient for me to avoid the issue so I am not blocked on anything.

**Comment by jgsogo** [comment]
Hi, @keef-cognitiv. We would like to fix this issue for next release 1.22 and we'd love to add some tests to avoid a regression. Do you know how can we reproduce this issue? Maybe using some docker image? Any hint regarding this will be very welcome.

Thanks!

**Comment by rarvidson** [comment]
We have been unable to move past 1.19 because of this issue as we based all our dev scripts in sh to make them portable. Most likely guess is that double quotes is needed to prevent globing and word splitting. The PS1 on my Ubuntu seems to trigger the issue and can be recreated with ease.

```
docker run -it --rm ubuntu:18.04
export PS1="\[\e]0;\u@\h: \w\a\]${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]\$"
sh
export CONAN_OLD_PS1=$PS1
```
produces the following output....
```
sh: 2: export: \w\a\]\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]$: bad variable name
```

**Comment by rarvidson** [comment]
Additionally, the shebang should be on the first line.

**Comment by rarvidson** [comment]
Just dived in a bit more to determine the PS1 generation. Appears it's the color_prompt in bash_rc that's causing it. So I was able to recreate this in docker like so...
```
docker run -it --rm -e force_color_prompt=yes ubuntu:18.04
export PS1
sh
export CONAN_OLD_PS1=$PS1
```
That's much more testable and reflective of the env.

**Comment by memsharded** [comment]
Hi @rarvidson 

This is good info, many thanks. If we can reproduce with this we might be able to fix it in next minor release, lets give it a try.

**Comment by memsharded** [comment]
Fixed in https://github.com/conan-io/conan/pull/6265, will be released in 1.22.1. 

Thanks very much!
